privilege-request-protocol.md

```diff
@@ -0,0 +1,208 @@
+associated in some way with the `Bearer` token.
+
+The method by which a server determines whether an HTTP client has sufficient
+privilege to access the requested resource is beyond the scope of this memo.
+Methods could include, but are not limited to, determining whether the
+authenticated identity is a member of a configured role group, or whether the
+user associated with the authenticated identity has endorsed the identified
+software application currently being used for the access attempted.
+
+The method by which a server determines whether it's appropriate to allow the
+client to request additional privilege is beyond the scope of this memo.
+Methods could include, but are not limited to, determining whether the
+authenticated identity is the controller of the resource, or is the owner of
+the server.
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL
+NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**NOT RECOMMENDED**",
+"**MAY**", and "**OPTIONAL**" in this document are to be interpreted as
+described in BCP 14 \[[RFC2119][]\] \[[RFC8174][]\] when, and only when, they
+appear in all capitals, as shown here.
+
+Operation
+---------
+The following is a typical exchange according to this protocol:
+
+1. The client attempts access to a resource.
+2. The server determines the client has insufficient privilege and denies the
+   request, returning an HTTP `403 Forbidden` response. The server further
+   determines it's appropriate to allow the client to request additional
+   privilege, so includes in the response a `Link` for the privilege request
+   endpoint URI.
+3. The client determines to request additional privilege and makes a simple
+   `POST` to the provided privilege request endpoint URI without credentials.
+4. The privilege request endpoint determines the request is acceptable and
+   queues the request, returning an HTTP `202 Accepted` response.
+
+The privilege management system might (but is not required to) provide a
+web-based permission management interface, and in some circumstances it might
+be appropriate to allow the client's user to access this interface. The
+privilege request endpoint **MAY** include in its response a URI [template][RFC6570]
+to access this interface, to which the client can redirect the user's trusted
+web browser. The interface URI template **SHOULD** include a `redirect_uri`
+variable so that the interface can eventually redirect back to the client
+application on completion of the management activity.
+
+On completion of the management activity (for example, but not limited to,
+in a permission management interface if provided, or offline by whatever means
+supported by the server) the client can attempt access again. There is no
+guarantee that sufficient privilege has been granted at this point, and new
+requests might still be denied, potentially restarting this process.
```

Comment:
will the client get notified somehow that their request has now been processed?

Reply:
if the was a permission management UI link given, then returning with the redirect_uri would indicate the management activity is concluded. if in the future we add a webhook (perhaps i'll add that for the next rev along with other changes), then the webhook could also be called.